util/geosop/GeomFunction.h

```diff
@@ -55,8 +54,43 @@ class Result {
     } typeCode;
 };
 
-typedef std::function<Result *( const std::unique_ptr<Geometry>&, const std::unique_ptr<Geometry>&, double d )>
-    geomFunSig;
+class Argument {
```

Comment:
I wonder if `Argument` and `Result` can be merged into a single `Value` type?  But Results own their data, whereas Arguments don't - so maybe not?